sql/codegen/xgboost/codegen.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xgboost
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/sql-machine-learning/sqlflow/sql/codegen"
+	"strings"
+)
+
+var attributeChecker = map[string]func(interface{}) error{
+	"model.eta": func(x interface{}) error {
+		switch x.(type) {
+		case float32, float64:
+			return nil
+		default:
+			return fmt.Errorf("model.eta should be of type float, received %T", x)
+		}
+	},
+	"model.num_class": func(x interface{}) error {
+		switch x.(type) {
+		case int, int32, int64:
+			return nil
+		default:
+			return fmt.Errorf("model.num_class should be of type int, received %T", x)
+		}
+	},
+	"train.num_boost_round": func(x interface{}) error {
+		switch x.(type) {
+		case int, int32, int64:
+			return nil
+		default:
+			return fmt.Errorf("train.num_boost_round should be of type int, received %T", x)
+		}
+	},
+	"model.objective": func(x interface{}) error {
+		if _, ok := x.(string); !ok {
+			return fmt.Errorf("model.objective should be of type string, received %T", x)
+		}
+		return nil
+	},
+}
+
+func resolveModelType(estimator string) (string, error) {
+	switch strings.ToUpper(estimator) {
+	case "XGBOOST.GBTREE":
+		return "gbtree", nil
+	case "XGBOOST.GBLINEAR":
```

Comment:
For the booster algorithms except `gbtree`, we need to specify it in parameters:
``` python
param = {'booster': 'dart',
         'max_depth': 5, 'learning_rate': 0.1,
```
ref: https://xgboost.readthedocs.io/en/latest/tutorials/dart.html#sample-script

Maybe we can only support the `gbtree` booster and support other booster algs with the unit tests in the future?

Reply:
@Yancey1989 The current code does support `param = {'booster': 'dart'}` on line 128 `params["model."]["booster"] = booster`.

Reply:
I see, thanks.